lib/nameMatcher.js

```diff
@@ -0,0 +1,307 @@
+var redis = require('redis');
+var natural = require('natural');
+var _ = require('underscore');
+
+var nameMatcher = {};
+var redisClient;
+// firstNames = e.g. {bob: bobMetaphoneHash, rachel: rachelMetaphoneHash}
+var firstNames;
+var lastNames;
+var firstNamesDev;
+var lastNamesDev;
+
+
+//static private helper functions
+/**
+ * Helper function to wrap standard callback-based async functions in a promise
+ * @param fn {function} The function to be called
+ * @param args {array} array of arguments to be passed to the function
+ * @param _this {object} (optional) object to use as `this` when calling the function defaults to null
+ * @return {Promise} promise to be resolved or rejected when the async fn is complete
+ */
+var promisify = function (fn, args, _this) {
+    _this = _this || null;
+    return new Promise(function promisifyWrapper(resolve, reject) {
+        var promiseCallback = function (err, result) {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(result);
+            }
+            return;
+        };
+        args.push(promiseCallback);
+        return fn.apply(_this, args);
+    });
+};
+
+
+/**
+ * Helper function to generate the metaphone for a name.
+ * Will retrieve the metaphone from the local collection of names if it exists.
+ * @param name {string} the name to generate a metaphone for
+ * @param localCollection {object} a collection of name-indexed metaphones (optional)
+ * @return {array} an array of the generated metaphones.
+ */
+var getMetaphone = function (name, localCollection) {
+    localCollection = localCollection || {};
+    var dm = natural.DoubleMetaphone;
+    if (localCollection[name]) {
+        return localCollection[name];
+    }
+    return dm.process(name);
+};
+
+
+/**
+ * simple function to clear the database of names
+ */
+nameMatcher.clearNames = function() {
+    redisClient.del('firstNames', console.log('deleted first'));
+    redisClient.del('lastNames', console.log('deleted last'));
+};
+
+
+/**
+ * Compare two names based on phonetics and string similarity
+ * @param name1 {string} the name to be used in the comparison
+ * @param name2 {string} the name to be compared against
+ * @param localCollection {object} a collection of name-indexed metaphones (e.g firstNames or lastNames)
+ * @return {boolean} true if names are considered a match. false otherwise
+ */
+var compareNames = function (name1, name2, localCollection) {
+    var jaro = natural.JaroWinklerDistance(name1, name2);
+    var mp1 = getMetaphone(name1, localCollection);
+    var mp2 = getMetaphone(name2, localCollection);
+    var match1 = mp1[0] === mp2[0] ? true : false;
+    var match2 = mp1[1] === mp2[1] ? true : false;
+    // Match the inputs
+    if (match1 === true &&
+        match2 === true &&
+        jaro >= 0.6) {
+        // Both metaphones match exactly, and strings have distance of 0.6
+        return true;
+    } else if ((match1 + match2) &&
```

Comment:
I agree, especially as it isnt clear unless you catch the assignment above that they are booleans